src/wal.rs

```diff
@@ -100,69 +94,244 @@ pub struct Wal {
 }
 
 impl Wal {
-    pub fn open(_path: PathBuf, _opts: AgateOptions) -> Result<Wal> {
-        unimplemented!()
+    /// open or create a WAL from options
+    pub fn open(path: PathBuf, opts: AgateOptions) -> Result<Wal> {
+        let (file, bootstrap) = if path.exists() {
```

Comment:
Seems need to check the file length.

Reply:
This simple WAL implementation do not have vlog header (which only includes encryption-related information). Thus we can directly start reading from it.

Reply:
What if the file is smaller than the given size?

Reply:
It seems that we do not have a pre-defined size for WAL. We just simply read from start to the end.